File: rust/prime_factors.rs
pub fn factors(mut n: u64) -> Vec<u64> {
    let mut arr = Vec::new();
    let mut i = 2;
    while n >= i {
        if n % i == 0 {
            arr.push(i);
            n /= i;
        } else {
            i += 1;
        }
    }
    arr
}

File: go/prime_factors.go
package prime

func Factors(n int64) []int64 {
	arr := []int64{}
	f := int64(2)
	for n >= f {
		if n%f == 0 {
			arr = append(arr, f)
			n /= f
		} else {
			f++
		}
	}
	return arr
}
